corehq/apps/hqwebapp/tables/elasticsearch/tables.py

```diff
@@ -133,14 +133,21 @@ def __len__(self):
     @property
     @memoized
     def _total_records(self):
-        res = self._get_es_results(self.query.size(0))
+        return self.get_total_records_in_query(self.query)
+
+    @classmethod
+    def get_total_records_in_query(cls, query):
+        """
+        Returns the total number of records in the ESQuery.
+        """
+        res = cls.get_es_results(query.size(0))
         if res is not None:
             return res['hits'].get('total', 0)
         else:
             return 0
 
     @staticmethod
-    def _get_es_results(query):
+    def get_es_results(query):
```

Comment:
I know that the combination of a "private method" with `staticmethod` doesn't make a ton of sense, but it seems like `ElasticTableData`'s purpose is mainly to allow iteration through `self.data` without knowing about the underlying mechanics of that data. Exposing elasticsearch functionality as a public method feels like it breaks that.

Reply:
noted, i'll explore a utility for htmx views and es query counts later when this code is in cleanup phase